### ISSUE TITLE
filter_by() got an unexpected keyword argument 'default'

### ISSUE THREAD

**Issue by lichtamberg** [saltstack/salt#23087]
Trying to use this formula https://github.com/saltstack-formulas/couchdb-formula and receive

```
filter_by() got an unexpected keyword argument 'default'
```

Working on:
OSX 10.9.5
Python 2.7.9

Error message:
```
    - Rendering SLS 'base:couchdb.install' failed: Jinja error: filter_by() got an unexpected keyword argument 'default'
    - /tmp/.deployer_85176d__salt/running_data/var/cache/salt/minion/files/base/couchdb/map.jinja(31):
    - ---
    - [...]
    -     }
    -   }
    - %}
    -
    - {# merge default values with the ones received in pillar.get #}
    - {% set couchdb  = salt['grains.filter_by'](conf_defaults, grain='None', merge=couchdb, default='couchdb') %}    <======================
    -
    - {% set merge_props = {
    -      'couchdb'  : {
    -        'source' :  'https://www.apache.org/dist/couchdb/source/' + couchdb.version + '/apache-couchdb-' + couchdb.version + '.tar.gz',
    -        'source_hash'   :  'https://www.apache.org/dist/couchdb/source/' + couchdb.version + '/apache-couchdb-' + couchdb.version + '.tar.gz.md5'
    - [...]
    - ---
    - Traceback (most recent call last):
    -   File "/usr/local/lib/python2.7/site-packages/salt/utils/templates.py", line 280, in render_jinja_tmpl
    -     output = template.render(**decoded_context)
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 969, in render
    -     return self.environment.handle_exception(exc_info, True)
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    -     reraise(exc_type, exc_value, tb)
    -   File "<template>", line 4, in top-level template code
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 1013, in make_module
    -     return TemplateModule(self, self.new_context(vars, shared, locals))
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 1070, in __init__
    -     self._body_stream = list(template.root_render_func(context))
    -   File "/tmp/.deployer_85176d__salt/running_data/var/cache/salt/minion/files/base/couchdb/map.jinja", line 31, in top-level template code
    -     {% set couchdb  = salt['grains.filter_by'](conf_defaults, grain='None', merge=couchdb, default='couchdb') %}
    - TypeError: filter_by() got an unexpected keyword argument 'default'
    -
    - Rendering SLS 'base:couchdb.user' failed: Jinja error: filter_by() got an unexpected keyword argument 'default'
    - /tmp/.deployer_85176d__salt/running_data/var/cache/salt/minion/files/base/couchdb/map.jinja(31):
    - ---
    - [...]
    -     }
    -   }
    - %}
    -
    - {# merge default values with the ones received in pillar.get #}
    - {% set couchdb  = salt['grains.filter_by'](conf_defaults, grain='None', merge=couchdb, default='couchdb') %}    <======================
    -
    - {% set merge_props = {
    -      'couchdb'  : {
    -        'source' :  'https://www.apache.org/dist/couchdb/source/' + couchdb.version + '/apache-couchdb-' + couchdb.version + '.tar.gz',
    -        'source_hash'   :  'https://www.apache.org/dist/couchdb/source/' + couchdb.version + '/apache-couchdb-' + couchdb.version + '.tar.gz.md5'
    - [...]
    - ---
    - Traceback (most recent call last):
    -   File "/usr/local/lib/python2.7/site-packages/salt/utils/templates.py", line 280, in render_jinja_tmpl
    -     output = template.render(**decoded_context)
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 969, in render
    -     return self.environment.handle_exception(exc_info, True)
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    -     reraise(exc_type, exc_value, tb)
    -   File "<template>", line 4, in top-level template code
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 1013, in make_module
    -     return TemplateModule(self, self.new_context(vars, shared, locals))
    -   File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 1070, in __init__
    -     self._body_stream = list(template.root_render_func(context))
    -   File "/tmp/.deployer_85176d__salt/running_data/var/cache/salt/minion/files/base/couchdb/map.jinja", line 31, in top-level template code
    -     {% set couchdb  = salt['grains.filter_by'](conf_defaults, grain='None', merge=couchdb, default='couchdb') %}
    - TypeError: filter_by() got an unexpected keyword argument 'default'
```

It looks like the port to OSX is not up to date 
```
# /usr/local/lib/python2.7/site-packages/salt/client/ssh/wrapper/grains.py:142
def filter_by(lookup_dict, grain='os_family', merge=None):
....
```

```
 ~/salt  ± master ●  pip show salt
---
Metadata-Version: 1.0
Name: salt
Version: 2014.7.5
Summary: Portable, distributed, remote execution and configuration management system
Home-page: http://saltstack.org
Author: Thomas S Hatch
Author-email: thatch45@gmail.com
License: Apache Software License 2.0
Location: /usr/local/lib/python2.7/site-packages
Requires: Jinja2, msgpack-python, PyYAML, MarkupSafe, requests
Entry-points:
  [console_scripts]
  salt = salt.scripts:salt_main
  salt-api = salt.scripts:salt_api
  salt-call = salt.scripts:salt_call
  salt-cloud = salt.scripts:salt_cloud
  salt-cp = salt.scripts:salt_cp
  salt-key = salt.scripts:salt_key
  salt-master = salt.scripts:salt_master
  salt-minion = salt.scripts:salt_minion
  salt-run = salt.scripts:salt_run
  salt-ssh = salt.scripts:salt_ssh
  salt-syndic = salt.scripts:salt_syndic
  salt-unity = salt.scripts:salt_unity
```

**Comment by lichtamberg** [comment]
I installed salt via PIP: didn't work (execption like above).
I installed salt via Homebrew: didn't work (execption like above).

**Comment by lichtamberg** [comment]
Ah ok.. This issue was fixed in version 2015.2.

Had to install saltstack via 
 
```
brew install saltstack --devel
```

And works fine now, this issue can be closed.
